src/module.js

```diff
@@ -41,15 +41,30 @@ class ModuleInterface {
 
   // returns Promise<string>
   source() {
+    // rk 2016-06-24:
+    // We should consider using lively.resource here. Unfortunately
+    // System.fetch (at least with the current systemjs release) will not work in
+    // all cases b/c modules once loaded by the loaded get cached and System.fetch
+    // returns "" in those cases
+
```

Comment:
Can't you invalidate the cache for `fetch`? Also, I would suggest that each module (optionally) has a resource. That way, read and write can be forwarded to the resource.